packages/components/src/lib/controllers/settings-controller.js

```diff
@@ -0,0 +1,264 @@
+    this._config = {
+    }
+  }
+
+  async init ({
+    api, store, configServiceUrl, clientId, appName, appVersion, branch, buildNumber, storageAdapter,
+    featureOptionsDefaults, resourceOptionsDefaults, uiOptionsDefaults, siteOptionsDefaults
+  } = {}) {
+    if (!api) {
+      throw new Error('API object is required for a settings controller initialization')
+    }
+    if (!store) {
+      throw new Error('Vuex store is required for a settings controller initialization')
+    }
+
+    this._api = api
+    this._store = store
+    this._storageAdapter = storageAdapter
+
+    this._configServiceUrl = configServiceUrl
+    this._clientId = clientId
+    this._appName = appName
+    this._appVersion = appVersion
+    this._branch = branch
+    this._buildNumber = buildNumber
+
+    this._featureOptionsDefaults = featureOptionsDefaults
+    this._resourceOptionsDefaults = resourceOptionsDefaults
+    this._uiOptionsDefaults = uiOptionsDefaults
+    this._siteOptionsDefaults = siteOptionsDefaults
+
+    const appConfigPromise = this.requestAppOptions()
+    const initOptionsPromise = this.initOptions()
+    /*
+    We do not allow user to configure lookupResourceOptions or siteOptions currently.
+    Because of this, they are initialized separately from the rest of the options.
+    `initOptions()` will be called several times during the app's lifetime,
+    after a user has been logged in, as one example.
+    `initNonConfigurableOptions()` will be called only once,
+    during an initialization of the Settings Controller.
+     */
+    this.initNonConfigurableOptions()
+    const [appConfigResponse] = await Promise.all([appConfigPromise, ...initOptionsPromise])
+    this._appConfig = await appConfigResponse.json() // Parse an app config's response into JSON
+
+    // region Public API of a settings controller
+    // A public API must be defined before modules are created because modules may use it
+    this._api.settings = {
+      initOptions: this.initOptions.bind(this),
+      getLexisOptions: () => this._appConfig && this._appConfig['lexis-cs'] ? this._appConfig['lexis-cs'] : {},
+      // TODO: logeion options are currently under the `lexis-cs` branch, at least in a dev config. Shall we fix it?
+      getLogeionOptions: () => this._appConfig && this._appConfig['lexis-cs'] && this._appConfig['lexis-cs'].logeion ? this._appConfig['lexis-cs'].logeion : {},
+      getFeatureOptions: this.getFeatureOptions.bind(this),
+      getResourceOptions: this.getResourceOptions.bind(this),
+      getUiOptions: this.getUiOptions.bind(this),
+      // we don't offer UI to change to lookupResourceOptions or siteOptions
+      // so they remain out of dynamic state for now - should eventually
+      // refactor
+      lookupResourceOptions: this._lookupResourceOptions, // TODO: This is not used at the moment, do we still need it?
+      siteOptions: this._siteOptions, // Site options seems to be not used right now
```

Comment:
Let's just remove siteOptions too. This was prototype code and I think if we ever offer that it will probably be done differently now anyway.

Reply:
I think I was wrong in this comment. It is used in LexicalQuery (the only place where it's used, actually): https://github.com/alpheios-project/alpheios-core/blob/dev-components-refactoring-7/packages/components/src/lib/queries/lexical-query.js#L395
So probably we have to keep it. What do you think?

Reply:
Oh, maybe it's never assigned. Will double check it tomorrow.

Reply:
well, then it seems we have left the code in there that enables us to use an alternate site-specific lexicon, but we just aren't actually setting it in the file. But ok, let's leave the siteOptions code in then just in case.

Reply:
OK